src/clients/Deluge.ts

```diff
@@ -0,0 +1,236 @@
+import axios, { AxiosResponse } from "axios";
+import { InjectionResult } from "../constants.js";
+import { CrossSeedError } from "../errors.js";
+import { Label, logger } from "../logger.js";
+import { Metafile } from "../parseTorrent.js";
+import { getRuntimeConfig } from "../runtimeConfig.js";
+import { Searchee } from "../searchee.js";
+import { TorrentClient } from "./TorrentClient.js";
+
+export default class Deluge implements TorrentClient {
+	private msgId = 0;
+	private loggedIn = false;
+	private delugeCookie = "";
+	private delugeWebUrl: URL;
+	private delugeLabel: string;
+	private enLabeled: Promise<boolean>;
+
+	constructor() {
+		this.delugeWebUrl = new URL(`${getRuntimeConfig().delugeWebUrl}`);
+		this.delugeLabel = getRuntimeConfig().dataCategory;
+	}
+
+	/**
+	 * validates the login and host for deluge webui
+	 */
+	async validateConfig(): Promise<void> {
+		const url = new URL(this.delugeWebUrl);
+		if (url.username && !url.password) {
```

Comment:
I believe this logic is best due to the way the delugeWebUrl needs to be structured to parse properly with URL() and redacting.

Usage is http://:password@localhost:8112 and conforms to URL's parsing password as the password, which leaves username undefined.

They should not be defining a username at all, and in most cases, I foresaw this as being a case of them forgetting the preceding ":" to distinguish that it is the password and not a username (as it would then be assumed as).

If they provide an user:pass - it will just use the password. So this seems to satisfy things the best way IMO.

Reply:
But if they forget the preceding colon in the link to specify it is a password to the URL constructor, password will be undefined and username will take the expected password value.

I was trying to prevent this misconfiguration/definition. I suppose it misses the case of neither being defined at all, though.

Technically, they should only be using http://:<password>@localhost:8112 as a url, with a preceding colon to conform to the constructor. We can either error out on anything else or add some handling for this.

Not sure about your preference.

Reply:
I think let's error just on no password. 
```suggestion
		if (!url.password) {
```

Reply:
Got it.